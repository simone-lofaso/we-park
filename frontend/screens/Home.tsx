import { Button } from '@rneui/themed';
import { SafeAreaView } from 'react-native';
import { RootStackScreenProps } from '../types';
import { ImageBackground, StyleSheet, Text, View } from "react-native";

const image = { uri: "https://reactjs.org/logo-og.png" };
import Splash from '../assets/images/splash.png';

// TODO: Real Home screen

export default function Home({ navigation }: RootStackScreenProps<'Home'>) {
  return (
    <View style={styles.container}>
      <ImageBackground source={Splash} resizeMode="cover" style={styles.image}>
      <SafeAreaView>
        <Button
          title='Go To Map'
          onPress={() => navigation.navigate('Map')}
        />
        <Button
          title='Go To Register'
          onPress={() => navigation.navigate('Register')}
        />
        <Button
          title='Go To ScanAI'
          onPress={() => navigation.navigate('ScanAI')}
        />
        <Button
          title='Go To Login'
          onPress={() => navigation.navigate('Login')}
        />
      </SafeAreaView>
      </ImageBackground>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  image: {
    flex: 1,
    justifyContent: "center"
  }
});
