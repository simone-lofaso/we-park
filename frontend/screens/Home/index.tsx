import { AntDesign } from '@expo/vector-icons';
import { useEffect, useState } from 'react';
import { SafeAreaView, StyleSheet, View } from 'react-native';
import { doPark } from '../../api';
import { Button } from '../../components';
import { green } from '../../constants/Colors';
import { RootStackScreenProps, User } from '../../types';
import { getUser } from '../../util';
import CoinCounter from './CoinCounter';

export default function Home({ navigation }: RootStackScreenProps<'Home'>) {
  const [user, setUser] = useState<User>();

  const startParkingPress = async () => {
    if (!user) return;
    if (user.tokens < 10) return;
    await doPark(user, navigation);
  };

  useEffect(() => {
    (async () => {
      const user = await getUser();
      if (user) setUser(user);
    })();
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.nav}>
        <CoinCounter
          numberOfCoins={user?.tokens || NaN}
          style={styles.currentCoins}
        />
        <AntDesign
          name='user'
          size={45}
          color='white'
          onPress={() => navigation.push('Profile')}
        />
      </View>
      <Button text='Start Parking' onPress={startParkingPress}>
        <CoinCounter numberOfCoins={-10} />
      </Button>
      <Button text='Scan (Parking)'>
        <CoinCounter numberOfCoins={10} />
      </Button>
      <Button text='Scan (Leaving)'>
        <CoinCounter numberOfCoins={10} />
      </Button>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: green,
    flex: 1,
    justifyContent: 'center',
  },
  currentCoins: {},
  nav: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    position: 'absolute',
    right: 0,
    top: 0,
  },
});
