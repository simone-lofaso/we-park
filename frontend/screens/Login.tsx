import { SafeAreaView, StyleSheet } from 'react-native';
import { RootStackScreenProps } from '../types';
import { green } from '../constants/Colors';
import { AuthForm } from '../components/AuthForm';

export default function Login({ navigation }: RootStackScreenProps<'Login'>) {
  return (
    <SafeAreaView style={styles.container}>
      <AuthForm formType='Login' navigation={navigation} />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: green,
    justifyContent: 'center',
    height: '100%',
    padding: 20,
    width: '100%',
  },
});
