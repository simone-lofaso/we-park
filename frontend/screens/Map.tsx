import { SafeAreaView, StyleSheet } from 'react-native';
import { RootStackScreenProps } from '../types';
import { Button, Text } from '@rneui/themed';

export default function Map({
  navigation,
  route,
}: RootStackScreenProps<'Map'>) {
  const { floor } = route.params.parkingSpace;
  const { garageName } = route.params;

  const handlePark = async () => {
    // TODO: connect api route to parking
    navigation.navigate('Home');
  };

  return (
    <SafeAreaView style={styles.container}>
      <Text>{`Please park on floor ${floor} in garage ${garageName}`}</Text>
      <Button title='I Parked!' onPress={handlePark} />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
    padding: 20,
    width: '100%',
  },
});
