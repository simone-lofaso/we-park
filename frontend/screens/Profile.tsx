import { Button, Input } from '@rneui/themed';
import { useState } from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
import Constants from 'expo-constants';
import { RootStackScreenProps } from '../types';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { deleteUser } from '../util';

export default function Profile({
  navigation,
}: RootStackScreenProps<'Profile'>) {
  const [plateNum, setPlateNum] = useState({ plateNum: '' });

  const handleSubmit = async () => {
    try {
      if (!plateNum.plateNum) return;
      const res = await fetch(
        `http://${
          Constants.expoConfig?.extra?.apiUrl || 'localhost'
        }:8000/api/v1/user/add-plate`,
        {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            ...plateNum,
            userId: await AsyncStorage.getItem('userId'),
          }),
        }
      );
      if (res.ok) {
        navigation.navigate('Home');
      }
    } catch (e) {
      console.error(e);
    }
  };

  return (
    <SafeAreaView style={styles.container}>
      <Input
        placeholder='Add License Plate'
        onChangeText={(e) => setPlateNum({ plateNum: e })}
      />
      <Button title='Add Plate' onPress={handleSubmit} />
      <Button
        title='Sign Out'
        onPress={async () => {
          await deleteUser();
          navigation.navigate('Login');
        }}
      />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'center',
  },
});
