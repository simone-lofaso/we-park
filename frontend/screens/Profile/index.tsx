import { Button } from '@rneui/themed';
import { useEffect, useState } from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
import { Text } from '../../components';
import { green } from '../../constants/Colors';
import { height, width } from '../../constants/Dimensions';
import { RootStackScreenProps, User } from '../../types';
import { deleteUser, getUser } from '../../util';
import SingleInputForm from './SingleInputForm';
import Constants from 'expo-constants';

export default function Profile({
  navigation,
}: RootStackScreenProps<'Profile'>) {
  const [user, setUser] = useState<User>();
  useEffect(() => {
    (async () => {
      const user = await getUser();
      if (!user) {
        navigation.navigate('Login');
        return;
      }
      setUser(user);
    })();
  }, []);

  const doDelete = async () => {
    if (!user) return;
    const res = await fetch(
      `http://${
        Constants.expoConfig?.extra?.apiUrl || 'localhost'
      }:8000/api/v1/user/delete`,
      {
        method: 'POST',
        body: JSON.stringify({ id: user.id }),
        headers: { 'Content-Type': 'application/json' },
      }
    );
    if (res.ok) {
      await deleteUser();
      navigation.navigate('Register');
    }
  };

  return (
    <SafeAreaView style={styles.container}>
      <Text>{user?.email}</Text>
      <SingleInputForm type='AddPlate' navigation={navigation} />
      <SingleInputForm type='ChangeEmail' navigation={navigation} />
      <SingleInputForm type='ChangePassword' navigation={navigation} />
      <Button
        title='Sign Out'
        onPress={async () => {
          await deleteUser();
          navigation.navigate('Login');
        }}
      />
      <Button title='Delete Account' onPress={doDelete} />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: green,
    display: 'flex',
    height: height,
    width: width,
  },
});
