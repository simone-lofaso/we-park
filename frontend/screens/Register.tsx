import { Button, Input, Text } from '@rneui/themed';
import { useState } from 'react';
import { ImageBackground, SafeAreaView, StyleSheet } from 'react-native';
import { doRegister } from '../api';
import NorthGarage from '../assets/images/north-garage.jpg';
import { RootStackScreenProps } from '../types';

type FormState = {
  email: string;
  password: string;
};

export default function Register({
  navigation,
}: RootStackScreenProps<'Register'>) {
  const [form, setForm] = useState<FormState>({
    email: '',
    password: '',
  });

  return (
    <SafeAreaView style={styles.container}>
      <ImageBackground source={NorthGarage} style={styles.picture}>
        <Text style={styles.register}>Register</Text>
        <Input
          placeholder='Email'
          onChangeText={(e) => setForm({ ...form, email: e })}
          inputStyle={styles.input}
          placeholderTextColor='white'
        />
        <Input
          placeholder='Password'
          onChangeText={(e) => setForm({ ...form, password: e })}
          inputStyle={styles.input}
          placeholderTextColor='white'
        />
        <Button title='Submit' onPress={() => doRegister(form, navigation)} />
        <Button
          title='Go to Login'
          onPress={async () => {
            navigation.navigate('Login');
          }}
        />
      </ImageBackground>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
    padding: 20,
    width: '100%',
  },
  picture: {
    alignItems: 'center',
    height: '100%',
    justifyContent: 'center',
    width: '100%',
  },
  register: {
    color: 'white',
    fontSize: 20,
  },
  input: {
    color: 'white',
  },
});
