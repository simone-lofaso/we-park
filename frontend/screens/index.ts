export { default as Home } from './Home';
export { default as Login } from './Login';
export { default as Map } from './Map';
export { default as Register } from './Register';
export { default as ScanAI } from './ScanAI';
export { default as Profile } from './Profile';
