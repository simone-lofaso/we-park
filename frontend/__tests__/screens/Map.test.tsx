import React from 'react';
import { create } from 'react-test-renderer';
import { Map } from '../../screens';

it(`renders correctly`, () => {
  const props: any = {
    route: {
      params: {
        parkingSpace: 1,
      },
    },
  };
  const tree = create(<Map {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
