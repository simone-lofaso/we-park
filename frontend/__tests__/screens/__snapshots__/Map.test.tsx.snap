// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "padding": 20,
      "width": "100%",
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#242424",
      }
    }
  >
    Please park on floor undefined in garage undefined
  </Text>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 2,
        },
        undefined,
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 2,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          I Parked!
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
